admin_solutions/authentication.adoc

```diff
@@ -5,10 +5,303 @@ toc::[]
+provider.
+
+[CAUTION]
+====
+Basic Authentication is not a secure exchange in principle; it needs to be
+combined with encryption or certificate authentication to prevent potential
+snooping of the user ID and password, and to prevent man-in-the-middle attacks.
+====
+
+With `*BasicAuthPasswordIdentityProvider*` configured, users send their username
+and password to {product-title}, which then validates those credentials against
+a remote server by making a server-to-server request, passing the credentials as
+a Basic Auth header. This requires users to send their credentials to
+{product-title} during login. 
+
+[NOTE]
+====
+This only works for username/password login mechanisms, and {product-title} must
+be able to make network requests to the remote authentication server. 
+====
+
+[[configuring-basic-auth-on-master]]
+=== Configuring Authentication on the Master
+
+. If you have:
+.. Already completed the installation of Openshift, then copy the
+*_/etc/origin/master/master-config.yaml_* file into a new directory; for example:
++
+----
+$ mkdir basicauthconfig; cp master-config.yaml basicauthconfig
+----
+.. Not yet installed {product-title}, then start the {product-title} API server,
+specifying the hostname of the (future) {product-title} master and a directory
+to store the configuration file created by the start command:
++
+----
+$ openshift start master --public-master=<apiserver> --write-config=<directory>
+----
++
+For example:
++
+----
+$ openshift start master --public-master=https://myapiserver.com:8443 --write-config=basicauthconfig
+----
++
+[NOTE]
+====
+If you are installing with Ansible, then you must add the
+`*identityProvider*` configuration to the Ansible playbook.
+If you use the following steps to modify your configuration manually after installing with Ansible, then you will lose any modifications whenever you re-run the install tool or upgrade.
+====
++
+. Edit the new *_master-config.yaml_* file's `*identityProviders*` stanza.
+. Copy
+xref:../install_config/configuring_authentication.adoc#basic-auth-example-config[the example `*BasicAuthPasswordIdentityProvider*` configuration] and paste it to replace the existing stanza.
+. Make the following modifications to the `*identityProviders*` stanza:
+.. Change the provider `*name*` ("my_remote_basic_auth_provider") to something
+unique and relevant to your deployment. This name is prefixed to the returned
+user ID to form an identity name.
+.. If required,
+xref:../install_config/configuring_authentication.adoc#mapping-identities-to-users[change `*mappingMethod*`] to control how mappings are established between the
+provider's identities and user objects.
+.. Specify the `*url*` to use to connect to a server that accepts credentials in Basic authentication headers.
+.. Optionally, change the `*ca*` to the certificate bundle to use in order to validate server certificates for the configured URL.
```

Comment:
none of the example sample paths are likely to work for them... I would say "set" instead of "change"